packages/vaadin-template-renderer/src/vaadin-template-renderer.js

```diff
@@ -0,0 +1,63 @@
+import { FlattenedNodesObserver } from '@polymer/polymer/lib/utils/flattened-nodes-observer.js';
+
+import './vaadin-template-renderer-templatizer.js';
+import { Templatizer } from './vaadin-template-renderer-templatizer.js';
+
+/**
+ * Returns the renderer function that renders the Polymer template to a root element.
+ * If the template has been appended to a root element once, then it only re-renders the template with the new properties.
+ */
+function createRenderer(template) {
+  const templatizer = Templatizer.create(template);
+
+  // TODO: properties VS model
+  return (root, properties) => {
+    templatizer.renderTo(root, properties);
+  };
+}
+
+/**
+ * Processes the Polymer template by creating an appropriate renderer function
+ * that is used to render this template.
+ */
+function processTemplate(component, template) {
+  component.renderer = createRenderer(template);
+}
+/**
+ * Detects and processes Polymer templates in the given component.
+ */
+function processTemplates(component) {
+  [...component.children]
+    .filter((child) => {
+      return child instanceof HTMLTemplateElement;
+    })
+    .filter((template) => {
+      // Exclude templates which have been processed earlier
+      if ('__templatizeOwner' in template) {
+        return false;
+      }
+
+      return true;
+    })
+    .forEach((template) => {
+      processTemplate(component, template);
+    });
+}
+
+/**
+ * Observes the children nodes of the component
+ * to process templates which are added or changed after the component's initialization.
+ */
+function observeTemplates(component) {
+  // The mutation observer attaches to a component node
+  // so it disconnects automatically as the component node is removed
```

Comment:
@tomivirkki Does it work in this way?

I decided so because, e.g the dialog component doesn't disconnect `FlattenedNodesObserver` in `disconnectedCallback`.